src/store/views.ts

```diff
@@ -76,5 +82,62 @@ export const useViewStore = defineStore('view', {
         });
       }
     },
+    serialize(stateFile: StateFile) {
+      const viewConfigStore = useViewConfigStore();
+      const { manifest } = stateFile;
+      const { views } = manifest;
+
+      // Serialize the layout
+      const serializeLayout = (
+        layout: Layout | string
+      ): StateFileLayout | string => {
+        if (typeof layout !== 'string') {
+          return {
+            direction: layout.direction,
+            items: layout.items.map((item) => serializeLayout(item)),
+          };
+        }
+        return layout;
+      };
+
+      manifest.layout = serializeLayout(this.layout) as StateFileLayout;
+
+      // Serialize the view specs
+      Object.entries(this.viewSpecs).forEach(([id, spec]) => {
+        const type = spec.viewType;
+        const { props } = spec;
+        const config = {};
+
+        const view = {
+          id,
+          type,
+          props,
+          config,
+        };
+
+        views.push(view);
+      });
+
+      // Serialize the view config
+      viewConfigStore.serialize(stateFile);
+    },
+    deserialize(views: View[], dataIDMap: Record<string, string>) {
```

Comment:
I noticed we don't restore the layout here. I think we will also need to store the name of the layout in the store as well, since we need to ensure the layout selector is correct.

Reply:
Oops, yes we are serializing it be never restoring it.